SamplesDashboard/SamplesDashboard/Services/NpmService.cs

```diff
@@ -30,18 +37,28 @@ public NpmService(IHttpClientFactory clientFactory)
         public async Task<string> GetLatestVersion(string packageName)
         {
             var httpClient = _clientFactory.CreateClient();
+         
             HttpResponseMessage responseMessage = await httpClient.GetAsync(string.Concat("https://registry.npmjs.org/", packageName));
 
             if (responseMessage.IsSuccessStatusCode)
             {
-                using (var stream = await responseMessage.Content.ReadAsStreamAsync())
-                using (var streamReader = new StreamReader(stream))
-                using (var jsonTextReader = new JsonTextReader(streamReader))
+                NpmQuery npmData;
+                if (!_cache.TryGetValue(responseMessage, out npmData))
```

Comment:
I think we want to avoid the network call entirely if we have the result in cache. Lets sync up on how we can do this better.